distributed/utils_test.py

```diff
@@ -144,36 +143,8 @@ async def cleanup_global_workers():
 
 
 @pytest.fixture
-def loop(cleanup):
-    with check_instances():
-        with pristine_loop() as loop:
-            # Monkey-patch IOLoop.start to wait for loop stop
-            orig_start = loop.start
-            is_stopped = threading.Event()
-            is_stopped.set()
-
-            def start():
-                is_stopped.clear()
-                try:
-                    orig_start()
-                finally:
-                    is_stopped.set()
-
-            loop.start = start
-
-            yield loop
-
-            # Stop the loop in case it's still running
-            try:
-                sync(loop, cleanup_global_workers, callback_timeout=0.500)
-                loop.add_callback(loop.stop)
-            except RuntimeError as e:
-                if not re.match("IOLoop is clos(ed|ing)", str(e)):
-                    raise
-            except asyncio.TimeoutError:
-                pass
-            else:
-                is_stopped.wait()
+def loop(loop_in_thread):
```

Comment:
This means `loop` is no longer calling `check_instances`. I'm not sure if that matters? I imagine a number of other fixtures that used `loop` were indirectly pulling in that check.

At this point, might as well just do `loop = loop_in_thread`?

Reply:
loop_in_thread calls cleanup which calls check_instances

https://github.com/dask/distributed/blob/24ea2d58b0b5e16da888e9d646bcbcaca5be2afa/distributed/utils_test.py#L1877

Reply:
Missed that, thanks!

Reply:
I think doing `loop = loop_in_thread` gives you two different fixtures with the same behavior, rather than one fixtures delegating to another